### PR TITLE
Adopt Black auto-formatting + provide optional pre-commit config, incl isort

### PR DESCRIPTION
**Pull request by stevejalim**
This changeset:

* adds Black and related deps to test requirements
* adds minimal Black config in pyproject.toml which will be used by default
* adds makefile commands to use black in check or write mode (`make checks` or `make autoformat`)
* updates CI config to run `black --check` in CI, too
* adds `.pre-commit-config.yaml` for pre-commit (see pre-commit.com) which runs `isort`, too
* has applied `black` formatting to the entire codebase
* bumps the versions of `Django` and `cryptography`, as per Dependabot prompts.

 - [X] Change has a jira ticket that has the correct status.
 - [X] Changelog entry added.
 - [X]  Requirements have been compiled.
 - [X]  Security patches/releases applied.
 - [X]  Reviewers may merge after approval

### CODE REVIEW

File: makefile [hunk 1]
```diff
@@ -10,6 +10,15 @@ install_requirements:
...
 	ENV_FILES='secrets-do-not-commit,dev' ./manage.py $(ARGUMENTS)
 
+autoformat:
+	# configuration for black is in pyproject.toml
+	black ./
+
+checks:
+	flake8 ./
+	# configuration for black is in pyproject.toml
+	black ./ --check
```
Comment (webbyfox) [inline makefile:20]: Do we need to set the line length as additional config or default is fine?

Comment (webbyfox) [reply makefile:20]: Ahh, I see pre-commit args. 120 which is :+1:

**Review by webbyfox — APPROVED** [review]
:100:

File: makefile (hunk 1 above)
Comment (stevejalim) [reply makefile:20]: Great question!

By default, Black reads config from `pyproject.toml` - https://github.com/psf/black#pyprojecttoml - which is what our IDEs/editors should auto-use too.

Comment (stevejalim) [reply makefile:20]: Pre-commit has its own args which am thinking we can probably skip/remove from its config

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -0,0 +1,39 @@
+repos:
+    - repo: https://github.com/psf/black
+      rev: 20.8b1
+      hooks:
+          - id: black
+          # Config for black lives in pyproject.toml
```
Comment (stevejalim) [inline .pre-commit-config.yaml:6]: @webbyfox fixed up the duplicated config - now there's only one place (`pyproject.toml`)

**Pull request merged**